banno-enrollment-po/BANNO.ENROLLMENT.V1.POW

```diff
@@ -0,0 +1,772 @@
+  POPUPMESSAGE(0,"Password Successfully Blanked Out")
+END [PROCEDURE]
+
+PROCEDURE CREATEEDITDIALOG
+[* Dialog that prompts user for information needed when creating
+** or editing a user.
+*]
+ DIALOGSTART("Create/Edit Banno User",200%,1)
+ IF NEWEMAILADDR<>"" THEN
+  DIALOGPROMPTCHAR("Email : ",40,NEWEMAILADDR)
+ ELSE
+  DIALOGPROMPTCHAR("Email : ",40,PREFEMAILADDR(SELECTEDPREF))
+ IF NEWUSERNAME<>"" THEN
+  DIALOGPROMPTCHAR("User Name : ",GETDATANUMBER(GETPARAMHBUSERNAMEMAXLENGTH),NEWUSERNAME)
+ ELSE
+  DIALOGPROMPTCHAR("User Name : ",GETDATANUMBER(GETPARAMHBUSERNAMEMAXLENGTH),PREFUSER(SELECTEDPREF))
+ IF ACTIONREQUESTED=ACTIONCREATEUSR THEN
+  DO
+   IF NEWPASSWORD<>"" THEN
+    DIALOGPROMPTCHAR("Enter Password:",GETDATANUMBER(GETPARAMHBPASSWORDMAXLENGTH),NEWPASSWORD)
+   ELSE
+    DIALOGPROMPTCHAR("Enter Password:",GETDATANUMBER(GETPARAMHBPASSWORDMAXLENGTH),"")
+  END
+ ELSE IF ACTIONREQUESTED=ACTIONEDITUSER  THEN
+  DIALOGINTROTEXT("Use Change Password Option to change Password")
+ IF ACTIONREQUESTED=ACTIONEDITUSER AND PREFPASSPROB(SELECTEDPREF)=0 THEN
+  DIALOGPROMPTYESNO("Change Password at next login?",FALSE)
+ ELSE IF ((ACTIONREQUESTED=ACTIONEDITUSER AND PREFPASSPROB(SELECTEDPREF)=1) OR
+           ACTIONREQUESTED=ACTIONCREATEUSR) THEN
+  DIALOGPROMPTYESNO("Change Password at next login?",TRUE)
+ IF ALLOWXFERTOANY=TRUE AND PREFXFERANYLOC(SELECTEDPREF)<>0 OR ACTIONREQUESTED=ACTIONCREATEUSR THEN
+  DIALOGPROMPTYESNO("Enable Xfer To Any Acct?",TRUE)
+ ELSE IF ALLOWXFERTOANY=TRUE AND PREFXFERANYLOC(SELECTEDPREF)=0 THEN
+  DIALOGPROMPTYESNO("Enable Xfer To Any Acct?",FALSE)
+ DIALOGDISPLAY
+ NEWUSERNAME=ENTERCHARACTER("User Name : ",GETDATANUMBER(GETPARAMHBUSERNAMEMAXLENGTH),PREFUSER(SELECTEDPREF))
+ NEWEMAILADDR=ENTERCHARACTER("Email : ",40,PREFEMAILADDR(SELECTEDPREF))
+ IF ACTIONREQUESTED=ACTIONCREATEUSR THEN
+  NEWPASSWORD=ENTERCHARACTER("Enter Password:",GETDATANUMBER(GETPARAMHBPASSWORDMAXLENGTH),"")
+ NEWPASSPROB=ENTERYESNO("Change Password at next login?","")
+ IF ALLOWXFERTOANY=TRUE THEN
+  CREATEXFERTOANY=ENTERYESNO("Enable Xfer To Any Acct?","")
+ DIALOGCLOSE
+END [PROCEDURE]
+
+PROCEDURE CREATEEDITFM
+[* File maintenance and error handling for creating or editing
+** a user.
+*]
+ IF NEWPASSPROB="Y" THEN
+  NEWPASSPROBNUM=1
+ ELSE
+  NEWPASSPROBNUM=0
+
+ IF ACTIONREQUESTED=ACTIONCREATEUSR THEN
+  DO
+   NEWPREFLOC=0
+   FMPERFORM CREATE PREFERENCE LOC AFTERLAST (CHECKPRIVS,0,NEWPREFLOC,FMERROR)
+    DO
+     SET HBUSERNAME TO UPPERCASE(NEWUSERNAME)
+     SET HBPASSWORD TO NEWPASSWORD
+     SET LASTHBPWCHANGEDATE TO SYSTEMDATE
+     SET PWDRESETPROBATION TO NEWPASSPROBNUM
+     SET EMAILADDRESS TO NEWEMAILADDR
+    END
+   IF NEWPREFLOC<>0 AND ALLOWXFERTOANY=TRUE AND
+      CREATEXFERTOANY="Y" THEN
+    DO
+     FMPERFORM CREATE PREFERENCE LOC NEWPREFLOC ACCESS LOC AFTERLAST (CHECKPRIVS,0,FMERROR)
+      DO
+       SET ACCESSTYPE TO 3
+      END
+    END
+   IF FMERROR<>"" THEN
+    CALL ERRORHANDLER
```

Comment:
Line 438: Check for FMERROR=”” unnecessary. Same in other procedure(s)

Reply:
Made one check at end of procedure. Same in create/edit cross account procedure.